packages/react-components/react-checkbox/src/components/Checkbox/useCheckbox.tsx

```diff
@@ -69,7 +70,10 @@ export const useCheckbox_unstable = (props: CheckboxProps, ref: React.Ref<HTMLIn
     },
     root: resolveShorthand(props.root, {
       required: true,
-      defaultProps: nativeProps.root,
+      defaultProps: {
+        ref: useFocusWithin<HTMLSpanElement>(),
```

Comment:
The story with `createCustomFocusIndicatorStyle()` & `useFocusWithin()` looks a bit dangerous for me as the dependency on `useFocusWithin()` is implicit. Do you have something on your mind to make it explicit?

Reply:
tbh I am not happy at all that this `focus-within` requires an extra hook, but globally there is no way to know what parent of the focused element should have this class.

The 'safety' mechanism currently is that `createCustomFocusIndicatorStyles` needs explicit `focus-within` selector to be used. The styles make sure that there is no any focus styles are applied unless `useFocusWithin` is called.

If anyone uses `:focus-within` directly (I think inputs do it) then they should know that there is no keyboard/mouse heuristic there.

If you have any proposals, I'd be happy to hear them, I'm not 100% happy with the current situation either

Reply:
I don't have a good proposal, the one that I have will be a regression to the previous behavior in terms of performance...